### PR TITLE
adding ability to set deployment extra env variables for things like specifying the part number

### PR DESCRIPTION
**Pull request by MxNxPx**
PR purpose:
In order to change the podtato part via environment variable, the helm chart has been updated to reflect 

Fixes:
https://github.com/podtato-head/podtato-head/issues/140

Other info:
Updated readme and bumped up the helm chart version in the Chart.yaml from 0.1.0 to 0.1.1

### CODE REVIEW

**Comment by joshgav** [comment]
Thanks @MxNxPx! I like how you added a generic property for adding _any_ additional env vars, that will probably be helpful. Maybe in the future we could make a first-class `partNumber` property too which directly touches just the `PODTATO_PART_NUMBER` env var.

What do you think of naming the general property `env` (as in `.Vales.entry.env`) rather than `extraEnvs`? That would match the property name in a PodSpec and be a bit more obvious to users. Plus it's shorter but still clear, always nice.

Thanks again!

**Comment by joshgav** [comment]
One more thought, it would be great if you have time to add a test that uses alternate part numbers by setting these values, for example by using helm's `--set` parameter.

**Comment by MxNxPx** [comment]
> One more thought, it would be great if you have time to add a test that uses alternate part numbers by setting these values, for example by using helm's `--set` parameter.

@joshgav - thanks for your feedback. the PR has been updated to use "env" as the parameter rather than "extraEnvs" and added a test to set a part for the hat service (also bumped up the helm chart version number)

File: delivery/chart/test.sh [hunk 1]
```diff
@@ -24,12 +24,14 @@ if [[ -n "${github_token}" && -n "${github_user}" ]]; then
 fi
 
 if [[ -z "${RELEASE_BUILD}" ]]; then
-    # replace ghcr.io/podtato-head/body with ghcr.io/podtato-head/<github_user>/body for tests
+    # replace ghcr.io/podtato-head/body with ghcr.io/podtato-head/<github_user>/body for tests and test changing hat part number
     helm upgrade --install podtato-head ${this_dir} \
         --set "images.repositoryDirname=ghcr.io/${github_user:+${github_user}/}podtato-head" \
+        --set "hat.env[0].name=PODTATO_PART_NUMBER" --set "hat.env[0].value=02" \
```
Comment (joshgav) [inline delivery/chart/test.sh:30]: We could expand the `scripts/test_services.sh` script to look in the returned file and see that it's the expected one, but that would require some coordination between the two scripts; this is great for now. Thanks!

**Comment by joshgav** [comment]
LGTM, thank you @MxNxPx!

**Pull request merged**